Update Congrats page for multi plugins

lupus2k opened the pull request:
Fixes https://github.com/Automattic/dotcom-forge/issues/2722

## Proposed Changes
* Update design/copy for multiple plugins congrats page
* Update both for mobile and desktop
* Add a button between subtitle and the plugins cards.
* The CTA of the button link to `:slug/wp-admin/plugins.php`
* The new button has a new track event

| Desktop | Mobile |
| --- | --- |
| ![image](https://github.com/Automattic/wp-calypso/assets/402286/eae8c429-5d14-4091-a88e-d26c7c743d62) | ![image](https://github.com/Automattic/wp-calypso/assets/402286/22a8f1b4-29d4-41af-b8bd-de11a23a16f9) |

## Testing Instructions

* Confirm the design of the multiple plugins congrats page matches the Figma design.
Example URL: `http://calypso.localhost:3000/marketplace/thank-you/:slug?plugins=woocommerce-subscriptions%2Cwoocommerce-product-bundles`
Design: VLoPHWqcewxNKZYjjkDu3O-fi-1701%3A2254
Check the button discussion for updates: VLoPHWqcewxNKZYjjkDu3O-fi-2446:3296
* Confirm tracks event are logged when buttons/links are clicked
* Confirm track event for `Setup the plugins` button

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?

lupus2k committed:
Update Congrats page for multi plugins

candy02058912 committed:
Use thankYouHeaderBody

lupus2k committed:
Fix themes ThankYou to match ThankYouProps

paulopmt1 commented on client/my-sites/checkout/checkout-thank-you/marketplace/use-plugins-thank-you-data.tsx:
```diff
@@ -173,6 +176,33 @@ export function usePluginsThankYouData( pluginSlugs: string[] ): ThankYouData {
+					sendTrackEvent( 'calypso_plugin_thank_you_setup_plugins_click' );
+				} }
+			>
+				{ translate( 'Setup the plugins' ) }
```
I see we have `Use the plugins` in Figma, any special reason to use `Setup the plugins` instead?

candy02058912 replied:
Context: p1687873537334839/1687856481.217509-slack-CKZHG0QCR

paulopmt1 reviewed (commented):
Can we align the button like Figma and use a little smaller header font for mobile?

![image](https://github.com/Automattic/wp-calypso/assets/1044309/0047a6f5-16e4-41a1-b86e-ef8dde3cedf2)

I see two different console.log errors, the API call ones also appear in production. Should we tackle them in a separate issue?

<img width="1252" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1044309/bad4a999-e32e-4d1b-8e71-975ad7b216c7">

lupus2k committed:
Fix button align for mobile

lupus2k commented:
> Can we align the button like Figma and use a little smaller header font for mobile?

Fixed in [this commit](https://github.com/Automattic/wp-calypso/pull/78671/commits/2af20b74886a630281b9039bb83fedd8ff9fe984) on the parent branch, and rebased to this one. Thanks for noticing this

lupus2k commented:
> I see two different console.log errors, the API call ones also appear in production. Should we tackle them in a separate issue?

Hey, @paulo, I can't reproduce the first error, but the second is because it's trying to get the plugin information from the .org API, and it's also happening on production. 
Do you think we should create a follow-up for this?

candy02058912 commented:
> > I see two different console.log errors, the API call ones also appear in production. Should we tackle them in a separate issue?
> 
> Hey, @paulo, I can't reproduce the first error, but the second is because it's trying to get the plugin information from the .org API, and it's also happening on production. Do you think we should create a follow-up for this?

The first error seems like something to do with hot reload? (assuming it's the `undefined` error)

candy02058912 committed:
Send plugins data to tracks

candy02058912 reviewed (approved):
It's working nicely on my end! Events are tracked!
